cli/src/datafeed.ts

```diff
@@ -0,0 +1,116 @@
+import { command, option, subcommands } from 'cmd-ts'
+import { insertHandler as adapterInsertHandler } from './adapter'
+import { insertHandler as aggregatorInsertHandler } from './aggregator'
+import { IDatafeedBulk, IDatafeedBulkInsertElement, ReadFile } from './cli-types'
+import {
+  contractConnectHandler,
+  contractInsertHandler,
+  functionInsertHandler,
+  organizationListHandler,
+  reporterInsertHandler as delegatorReporterInsertHandler
+} from './delegator'
+import { insertHandler as listenerInsertHandler } from './listener'
+import { insertHandler as reporterInsertHandler } from './reporter'
+import { isValidUrl, loadJsonFromUrl } from './utils'
+
+export function datafeedSub() {
+  // datafeed bulk-insert --source ${source}
+
+  // TODOs
+  // datafeed bulk-remove --source ${source}
+  // datafeed bulk-activate --source ${source}
+  // datafeed bulk-deactivate --source ${source}
+
+  const insert = command({
+    name: 'bulk-insert',
+    args: {
+      data: option({
+        type: ReadFile,
+        long: 'source'
+      })
+    },
+    handler: bulkInsertHandler()
+  })
+
+  return subcommands({
+    name: 'datafeed',
+    cmds: { insert }
+  })
+}
+
+export function bulkInsertHandler() {
+  async function wrapper({ data }: { data }) {
+    const bulkData = data as IDatafeedBulk
+
+    const chain = bulkData?.chain || 'localhost'
+    const service = bulkData?.service || 'DATA_FEED'
+    const organization = bulkData?.organization || 'bisonai'
+    const functionName = bulkData?.functionName || 'submit(uint256, int256)'
+    const eventName = bulkData?.eventName || 'NewRound'
+    const organizationId = (await organizationListHandler()()).find(
+      (_organization) => _organization.name == organization
+    ).id
+
+    if (!checkBulkSource(data?.bulk)) {
+      console.error('invalid json src format')
+      return
+    }
+    for (const insertElement of bulkData.bulk) {
+      const adapterData = await loadJsonFromUrl(insertElement.adapterSource)
```

Comment:
I think we should try to catch if supplied path is incorrect, and decide whether we want to proceed with the rest. The current code would not display what has already been inserted and what not, causing problems when we try to insert many elements at once but not knowing from where to retry. Let's add some logs to show us progress of bulk inserts.

Reply:
I thought path checking was unnecessary since it's checked through line 54
`if (!checkBulkSource(data?.bulk)) {...`  

Reply:
I think it just check whether the path is valid, not whether there is some data there.

https://github.com/Bisonai/orakl/blob/88c68ca4763ba6007579ad3352fe8e0bd6ab6aec/cli/src/utils.ts#L40-L47

Reply:
I see